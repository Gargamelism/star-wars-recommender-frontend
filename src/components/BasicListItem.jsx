import React from "react";

export default function BasicListItem({ name }) {
    return (
        <span key={name} className="flex-grow text-center border-2 rounded border-gray-400 m-1 px-1">{name}</span>
    );
}