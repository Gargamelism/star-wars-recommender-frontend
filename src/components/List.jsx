import React from 'react';

export default function List ({ filterVal, listItem, items }) {
    
    const filteredItems = items.filter(({name}) => (
        name.toLowerCase().includes(filterVal.toLowerCase())
        ))
        .map((item) => (listItem(item)));



    return (
        <div className="flex justify-center w-9/12">
            {filteredItems.length > 0 ? filteredItems : 'The Force is weak with this search'}
        </div>
    );
}