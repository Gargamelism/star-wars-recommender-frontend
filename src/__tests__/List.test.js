import renderer from "react-test-renderer";
import BasicListItem from "../components/BasicListItem";
import List from "../components/List";

test("renders List", () => {
  const tree = renderer
    .create(<List filterVal="" listItem={BasicListItem} items={[{name: "123"}, {name: "345"}]} emptyListMsg="empty!" />)
    .toJSON();
    
    expect(tree).toMatchSnapshot();  
});
