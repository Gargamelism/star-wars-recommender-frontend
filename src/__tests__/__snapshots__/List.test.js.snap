// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List 1`] = `
<div
  className="flex flex-col justify-center items-center w-9/12 md:w-max md:w-1/2"
>
  <span
    className="flex-grow text-center border-2 rounded border-gray-400 m-1 px-1"
  >
    123
  </span>
  <span
    className="flex-grow text-center border-2 rounded border-gray-400 m-1 px-1"
  >
    345
  </span>
</div>
`;
